eo-runtime/src/test/java/org/eolang/DataizedLogLevelTest.java

```diff
@@ -0,0 +1,143 @@
+ * SOFTWARE.
+ */
+package org.eolang;
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+/**
+ * Test case for dataization log level.
+ *
+ * @since 0.22
+ */
+@Execution(ExecutionMode.CONCURRENT)
+public final class DataizedLogLevelTest {
+    @Test
+    public void shortLogs() throws InterruptedException {
+        final Logger log = Logger.getLogger(Dataized.class.getName());
+        final Level before = log.getLevel();
+        log.setLevel(Level.ALL);
+        final List<LogRecord> logs = new LinkedList<>();
+        final Handler hnd = new Handler() {
+            @Override
+            public void publish(final LogRecord record) {
+                logs.add(record);
+            }
+
+            @Override
+            public void flush() {
+                throw new UnsupportedOperationException("#flush()");
+            }
+
+            @Override
+            public void close() throws SecurityException {
+                throw new UnsupportedOperationException("#close()");
+            }
+        };
+        log.addHandler(hnd);
+        Thread thread = new Thread(() -> {
+            System.clearProperty("max.dataization.log");
+            System.getProperties().setProperty("max.dataization.log", String.valueOf(1));
+            final Phi phi = new PhiDec(Phi.Φ);
+            new Dataized(phi).take();
+        });
+        thread.start();
+        thread.join();
+        log.setLevel(before);
+        log.removeHandler(hnd);
+        MatcherAssert.assertThat(
+            logs.size(),
+            Matchers.equalTo(1)
+        );
+    }
+
+    @Test
+    public void longLogs() throws InterruptedException {
+        final Logger log = Logger.getLogger(Dataized.class.getName());
+        final Level before = log.getLevel();
+        log.setLevel(Level.ALL);
+        final List<LogRecord> logs = new LinkedList<>();
+        final Handler hnd = new Handler() {
+            @Override
+            public void publish(final LogRecord record) {
+                logs.add(record);
+            }
+
+            @Override
+            public void flush() {
+                throw new UnsupportedOperationException("#flush()");
+            }
+
+            @Override
+            public void close() throws SecurityException {
+                throw new UnsupportedOperationException("#close()");
+            }
+        };
+        log.addHandler(hnd);
+        Thread thread = new Thread(() -> {
+            System.clearProperty("max.dataization.log");
+            System.getProperties().setProperty("max.dataization.log", String.valueOf(2));
```

Comment:
@Karina5005 I believe we want to change the property to whatever it was before the test.

Reply:
fixed

Reply:
@Karina5005 Didn't get what has changed here: I meant that it would be good to reset max dataization log level to default value after the test.

Reply:
@mximp Max dataization log level depends on the property, I returned it to its original state. Nothing has changed in the existing threads, but in the new ones, the value that was originally passed (or default) will be set.